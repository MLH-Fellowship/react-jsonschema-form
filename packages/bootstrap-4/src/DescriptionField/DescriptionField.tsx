import React from "react";
import { FieldProps } from "@rjsf/core";

// export interface DescriptionFieldProps extends Partial<FieldProps> {
//   description?: string;
// }

const DescriptionField = ({ description }: FieldProps) => {
  if (description) {
    return <h5 className="mb-2">{description}</h5>;
  }

  return null;
};

export default DescriptionField;
