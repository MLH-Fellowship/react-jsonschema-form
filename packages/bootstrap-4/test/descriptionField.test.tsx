import React from "react";
import DescriptionField from "../src/DescriptionField";
import { JSONSchema7 } from "json-schema";
import renderer from "react-test-renderer";

describe("DescriptionField", () => {
  const schema: JSONSchema7 = {
    type: "array",
    items: {
      type: "string",
    },
  };
  const mockHandler = (): void => void 0;
  test("should return null when no description as a props is passed", () => {
    const tree = renderer
      .create(
        <DescriptionField
          uiSchema={{}}
          idSchema={{
            $id: "one",
          }}
          formData={{}}
          errorSchema={{}}
          registry={{
            fields: {},
            widgets: {},
            definitions: {},
            formContext: {},
          }}
          name="name 1"
          schema={schema}
          required
          disabled={false}
          readonly
          autofocus
          label="Some simple label"
          onChange={mockHandler}
          onBlur={mockHandler}
          onFocus={mockHandler}
          multiple={false}
          rawErrors={[""]}
          value="value"
          options={{}}
          type="text"
          formContext={{}}
          id="_id"
        />
      )
      .toJSON();
    expect(tree).toBe(null);
  });

  test("should return h2 element when description is being passed as props", () => {
    const tree = renderer
      .create(
        <DescriptionField
          uiSchema={{}}
          idSchema={{
            $id: "one",
          }}
          formData={{}}
          errorSchema={{}}
          registry={{
            fields: {},
            widgets: {},
            definitions: {},
            formContext: {},
          }}
          name="name 1"
          schema={schema}
          required
          disabled={false}
          readonly
          autofocus
          label="Some simple label"
          onChange={mockHandler}
          onBlur={mockHandler}
          onFocus={mockHandler}
          multiple={false}
          rawErrors={[""]}
          value="value"
          options={{}}
          type="text"
          formContext={{}}
          id="_id"
          description="SOME THING"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
