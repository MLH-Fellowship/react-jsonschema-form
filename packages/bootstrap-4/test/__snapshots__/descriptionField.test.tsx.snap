// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionField should return h2 element when description is being passed as props 1`] = `
Array [
  <style
    type="text/css"
  >
    
    .h2 {
        margin-top: 5px
    }

    
  </style>,
  <h2>
    SOME THING
  </h2>,
]
`;
